scripts/cibuild.sh

```diff
@@ -0,0 +1,12 @@
+set -e
+echo "Running nose tests"
+nosetests -w lib/
+echo "Running Ansible-lint"
+ansible-lint -x 204,301
+echo "Ansible-lint returned OK"
+echo "Running flake8 to check Python code"
+flake8
+echo "flake8 returned OK"
+echo "Checking for files which have no newline at end of file"
+git ls-files | tr '\n' '\0' | xargs -0 -L1 bash -c 'if test "$(grep --files-with-matches --binary-files=without-match --max-count=1 --regexp='.*' "$0")" && test "$(tail --bytes=1 "$0")"; then echo "No new line at end of $0."; false; fi'
```

Comment:
This quoting doesn't look quite correct, because it uses single quotes around the arg to bash, but then uses a single quote inside that around that for the `--regexp='.*'`

It'd probably be better to put this in a separate script, and have `xargs` execute it directly as:

```suggestion
git ls-files | tr '\n' '\0' | xargs -0 -L1 /usr/bin/bash "$(dirname "$0")/checkMissingEOF"
```

Also, not sure what the difference between `-L1` and `-n1` is here. I've never used `-L` with xargs.

In any case, whatever scripts you create should pass through `shellcheck`.

Reply:
Good point. I did run shellcheck though I must admit I glossed over the report about incorrect quotes. I will do a thorough pass again and consider your suggestion to refactor into a smaller script to keep the command line manageable.

Reply:
Mainly, I don't think shellcheck will be able to evaluate the quoted arg to bash.

Reply:
Thanks Christopher. I pulled the EOF newline check into a separate script, cleaned things up a bit and ensured shellcheck passes (a cool thing it flagged in the process was SC2143 which I have addressed by using grep -q). Thanks again for pushing on these, I am learning a bunch about scripting good practices in the process.